app/db.py

```diff
@@ -378,8 +380,9 @@ class Subnode:
     the Agora root) in the attribute 'uri'."""
     def __init__(self, path, mediatype='text/plain'):
         # Use a subnode's URI as its identifier.
-        self.uri = path_to_uri(path)
-        self.url = '/subnode/' + path_to_uri(path)
+        self.uri: str = path_to_uri(path)
+        self.url = '/subnode/' + self.uri
+        self.wlmeta = self.uri.split('/')[-1]
```

Comment:
I tend to make crappy acronyms, this indeed stands for wikilink meta, it needs to include the `.md`. I thought about a name for half a second and then said sure. I will fix this.

Reply:
I'm sorry, I don't quite understand what you mean here

Perhaps the first paragraph is a suggestion of what to name wlmeta

The second one I just don't understand. I do want the last part of the url, the `pagename.md`. 

This should eventually be provided in a normal way (hack hack), but the router is what gets the query string and it is never passed along. I thought about fixing that but then decided it was not worth the effort, at least in this PR

Reply:
If this is working for you we can merge it this way and then iterate, but aren't you passing this as path to https://github.com/boehs/oasis/edit/master/{path}? I don't get why you want pagename.md instead of path/to/pagename.md; your repo has markdown files at the top level, but other repos might not (I keep my journal files in /journal/, for example) :)

Reply:
Oh God your right 👍

Reply:
Oh God your right 👍